From b6441f732a3e6fd81f9f6b0ceb4f36866340f154 Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Mon, 15 Jan 2024 13:19:24 +0100
Subject: [PATCH 35/55] netifd: only reload swconfig if something changed

prevent swconfig switches from being reconfigured when the config didn't actually change.

Signed-off-by: John Crispin <john@phrozen.org>
---
 package/network/config/netifd/files/etc/init.d/network | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/package/network/config/netifd/files/etc/init.d/network b/package/network/config/netifd/files/etc/init.d/network
index b8b2a219b1..11bf3eabdf 100755
--- a/package/network/config/netifd/files/etc/init.d/network
+++ b/package/network/config/netifd/files/etc/init.d/network
@@ -9,7 +9,11 @@ init_switch() {
 	setup_switch() { return 0; }
 
 	include /lib/network
-	setup_switch
+	uci show network | grep @switch | sort | md5sum | cut -d" " -f1 > /var/run/swconfig.new
+	[ "$(cat /var/run/swconfig.new)" != "$(cat /var/run/swconfig.applied)" ] && {
+		mv /var/run/swconfig.new /var/run/swconfig.applied
+		setup_switch
+	}
 }
 
 start_service() {
-- 
2.34.1

